activerecord/lib/active_record/timestamp.rb

```diff
@@ -83,6 +83,17 @@ def timestamp_attributes_for_update
         def current_time_from_proper_timezone
           default_timezone == :utc ? Time.now.utc : Time.now
         end
+
+        def _insert_attributes(attributes)
+          super
+
+          attributes.map do |insert_attribute|
+            current_time = current_time_from_proper_timezone
```

Comment:
This gives us a new time for every row in the database which is wrong, because the eventual insert is one statement.